plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/packagejson/N4JSProjectSetupJsonValidatorExtension.xtend

```diff
@@ -1405,7 +1410,7 @@ public class N4JSProjectSetupJsonValidatorExtension extends AbstractPackageJSONV
 						val NodeModulesFolder nmFolder = nodeModulesDiscoveryHelper.getNodeModulesFolder(projectPath);
 						if (nmFolder !== null) {
 							val projectName = new EclipseProjectName(project.name).toN4JSProjectName
-							res.put(projectName, nmFolder.nodeModulesFolder);
+							res.putAll(projectName, nmFolder.nodeModulesFolders);
```

Comment:
The map will contain the global node_modules folder (the one in the root folder of the yarn workspace) many times: 1 + n (if there are n projects in the yarn workspace). Does this make sense?

Reply:
This is for caching purposes only since the related yarn/npm node_modules folders should not be computed everytime one of the six check methods is called. Regarding memory there might be a smarter way to cache it, I agree.